src/player/systems/attacks.rs

```diff
@@ -11,11 +11,18 @@ use thetawave_interface::{
 
 use crate::spawnable::{InitialMotion, SpawnProjectileEvent};
 
-/// Increase fire rate of player based on the amount of money collected
-// TODO: Remove hardcoded values
-pub fn player_scale_fire_rate_system(mut player_query: Query<&mut PlayerComponent>) {
-    for mut player in player_query.iter_mut() {
-        player.fire_period = 1.0 / (1.5 * ((0.8 * player.money as f32) + 4.0).ln());
```

Comment:
If not this implementation, then we should use an EventReader to update this `player.fire_period` field when the player collects money.